### PR TITLE
[FEATURE] Localize Container in Copy-Mode #12

### PR DESCRIPTION
**Pull request by achimfritz**
Localize a container in Copy-Mode (copyToLanguage) should also localize the children in Copy-Mode and remap the tx_container_parent field

### CODE REVIEW

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * @var Database
+     */
+    protected $database = null;
+
+    /**
+     * @param ContainerFactory|null $containerFactory
+     * @param Database|null $database
+     */
+    public function __construct(Database $database = null)
```
Comment (ervaude) [inline Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php:29]: Annotations not needed and also wrong

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * @param DataHandler $dataHandler
+     */
+    public function processCmdmap_afterFinish(DataHandler $dataHandler): void
+    {
+        $cmdmap = $dataHandler->cmdmap;
+        $copyMappingArray_merged = $dataHandler->copyMappingArray_merged;
+
+        foreach ($cmdmap as $table => $_) {
+            foreach ($cmdmap[$table] as $id => $incomingCmdArray) {
```
Comment (ervaude) [inline Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php:43]: Isn't `$cmdmap[$table]` the value of the first foreach (`$_`)?

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                $newId = $copyMappingArray_merged['tt_content'][$id];
+                $childrenInCopiedLanguage = $this->database->fetchRecordsByParentAndLanguage($id, $copyToLanguage);
+                $data = [
+                    'tt_content' => []
+                ];
+                foreach ($childrenInCopiedLanguage as $child) {
+                    $data['tt_content'][$child['uid']] = ['tx_container_parent' => $newId];
+                }
+
+                if (!empty($data['tt_content'])) {
```
Comment (ervaude) [inline Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php:66]: Would invert this conditions to a `continue` statement as conditions above

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    public function __construct(Database $database = null)
+    {
+        $this->database = $database ?? GeneralUtility::makeInstance(Database::class);
+    }
+
+    /**
+     * @param DataHandler $dataHandler
+     */
+    public function processCmdmap_afterFinish(DataHandler $dataHandler): void
```
Comment (ervaude) [inline Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php:37]: Anotations not needed

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        $copyMappingArray_merged = $dataHandler->copyMappingArray_merged;
+
+        foreach ($cmdmap as $table => $_) {
+            foreach ($cmdmap[$table] as $id => $incomingCmdArray) {
+                if (!is_array($incomingCmdArray)) {
+                    continue;
+                }
+                if ($table !== 'tt_content') {
+                    continue;
+                }
```
Comment (ervaude) [inline Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php:49]: This condition should be moved to the outer loop

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php (hunk 3 above)
Comment (achimfritz) [reply Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php]: fixed

File: Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php (hunk 5 above)
Comment (achimfritz) [reply Classes/Hooks/Datahandler/CommandMapAfterFinishHook.php]: fixed

**Pull request closed without merging**